### PR TITLE
Log empty or invalid parents array for documents

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- First step of https://github.com/dust-tt/tasks/issues/1661.
- Add logging on empty parents upon upsertion in both `core` and `front`.
- Add logging on `parents[0] != document_id` upon upsertion in both `core` and `front`.
- Related [MB query](https://metabase.dust.tt/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjozLCJ0eXBlIjoibmF0aXZlIiwibmF0aXZlIjp7InF1ZXJ5Ijoic2VsZWN0IFRPX0NIQVIqIGZyb20gZGF0YV9zb3VyY2VzX2RvY3VtZW50cyBXSEVSRSBzdGF0dXMgPSAnbGF0ZXN0JyBBTkQgcGFyZW50cyA9ICd7fScgT1JERVIgQlkgaWQgREVTQyBMSU1JVCAxMDA7IiwidGVtcGxhdGUtdGFncyI6e319fSwiZGlzcGxheSI6InRhYmxlIiwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnt9fQ==).

## Risk

Low.

## Deploy Plan

- Deploy `core`
- Deploy `front`

### CODE REVIEW

**Review by Fraggle — APPROVED** [review]

File: front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]/index.ts [hunk 1]
```diff
@@ -472,6 +475,25 @@ async function handler(
...
+      ];
+      if (!r.data.parents || r.data.parents.length === 0) {
+        statsDClient.increment(
+          "document_without_a_parent.count",
+          1,
+          statsDTags
+        );
+      } else if (r.data.parents[0] != req.query.documentId) {
+        statsDClient.increment(
+          "document_without_a_parent.count",
```
Comment (spolu) [inline front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]/index.ts:491]: Wrong name ?

**Review by spolu — COMMENTED** [review]
Awesome. One metric name may be off?

File: front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]/index.ts (hunk 1 above)
Comment (aubin-tchoi) [reply front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]/index.ts:491]: yes, copy pasted from above 🤦 
fixed in https://github.com/dust-tt/dust/pull/9013

Comment (spolu) [reply front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]/index.ts:491]: 👍

**Pull request merged**